projet/projects/views.py

```diff
@@ -103,3 +103,151 @@ def get_project_pending_contributors(request, project_id, statum):
             )
             serializer = ContributorSerializer(accepted_contributors, many=True)
             return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class CreateFeature(APIView):
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request, project_id, format=None):
+        project = Project.objects.get(id=project_id)
+
+        if (
+            Contributor.objects.filter(
+                user=request.user, project=project, accepted=True
+            ).exists()
+            is not True
+            and project.user != request.user
+        ):
+            error = "You have to be an accepted contributor or the project manager before you can make contributions."
+            return Response(error, status=status.HTTP_401_UNAUTHORIZED)
+
+        data = request.data
+        # _mutable = data._mutable
+
+        # # set to mutable
+        # data._mutable = True
+
+        # # сhange the values you want
+
+        # # set mutable flag back
+
+        data["project"] = str(project.id)
+        data["contributor"] = str(request.user.id)
+        # data._mutable = _mutable
+        serializer = FeatureSerializer(data=data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        else:
+            error = "Could not save feature"
+            return Response(error, status=status.HTTP_400_BAD_REQUEST)
+        error = "Method not allowed"
+        return Response(error, status=status.HTTP_405_METHOD_NOT_ALLOWED)
+
+
+class FeatureDetail(generics.RetrieveUpdateDestroyAPIView):
+    serializer_class = FeatureSerializer
+
+    def get_queryset(self):
+        return Feature.objects.filter(contributor=self.request.user)
+
+
+@api_view(["DELETE"])
+def delete_feature(request, feature_id=None):
+    if request.method == "DELETE":
+        try:
+
+            feature = Feature.objects.get(id=feature_id)
+
+            project = feature.project
+            if request.user == feature.contributor or request.user == project.user:
+                feature.delete()
+                return Response(
+                    f"{feature.name} has been deleted from {project.title}",
+                    status=status.HTTP_200_OK,
+                )
+
+            error = "You have to be the owner of this feature or manager of this project before you can delete."
+            return Response(error, status.HTTP_401_UNAUTHORIZED)
+        except Feature.DoesNotExist:
+            error = "Feature does not exist."
+            return Response(error, status=status.HTTP_404_NOT_FOUND)
+
+
+@api_view(["PATCH"])
+def review_feature(request, feature_id, format=None):
```

Comment:
Review feature allows managers to review a feature to make it ready to be approved.